api/v1alpha1/buildkit_types.go

```diff
@@ -0,0 +1,84 @@
+// Copyright 2025 SeatGeek, Inc.
+//
+// Licensed under the terms of the Apache-2.0 license. See LICENSE file in project root for terms.
+
+package v1alpha1
+
+import (
+	"github.com/reddit/achilles-sdk-api/api"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	TypeDeployed api.ConditionType = "Deployed"
+)
+
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:shortName=buildkit
+// +kubebuilder:subresource:status
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:printcolumn:name="Template",type=string,JSONPath=`.spec.template`
+type Buildkit struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   BuildkitSpec   `json:"spec,omitempty"`
+	Status BuildkitStatus `json:"status,omitempty"`
+}
+
+type BuildkitSpec struct {
+	// Template is the name of the BuildkitTemplate to use for creating the Buildkit instance.
+	// +kubebuilder:validation:Required
+	Template string `json:"template"`
+
+	// Resources defines the resource requirements for the Buildkit instance.
+	// It is optional and can be omitted if the default resource limits are sufficient.
+	Resources *corev1.ResourceRequirements `json:"resources,omitempty"`
```

Comment:
Should we use the `// +optional` marker here?

Reply:
Implemented via 669cec8